src/sources/aws_ecs_metrics/parser.rs

```diff
@@ -258,65 +258,102 @@ fn cpu_metrics(
     namespace: &Option<String>,
     tags: &BTreeMap<String, String>,
 ) -> Vec<Metric> {
-    let mut metrics = vec![gauge(
-        "cpu",
-        "online_cpus",
-        namespace.clone(),
-        timestamp,
-        cpu.online_cpus as f64,
-        tags.clone(),
-    )];
+    let mut metrics = vec![];
```

Comment:
Nit: you could make this `Vec::with_capacity(<number of expected metrics>)` to avoid reallocations below.